src/protocol/handshake.rs

```diff
@@ -24,6 +24,31 @@ enum NodeType {
     Private = 32,
 }
 
+/// Handshake configuration allows some customization of the handshake procedure.
+#[derive(Clone)]
+pub struct HandshakeCfg {
+    /// Will flip a random bit in a random byte of shared value used for session signing.
+    pub bit_flip_shared_val: bool,
+
+    /// Will flip a random bit in a random byte of the public key.
+    pub bit_flip_pub_key: bool,
+
```

Comment:
matter of taste here, but I would go with one less underscore and treat bitflip as a one unit.  I think it might improve readability later on in some of the longer lines & variable names.
```
   pub bitflip_shared_val: bool,

   /// Will flip a random bit in a random byte of the public key.
   pub bitflip_pub_key: bool,
```

Reply:
nice, I like it! thanks.
Done